src/app/pages/review-attributes/_services/review-attributes.model.service.ts

```diff
@@ -1,33 +1,59 @@
-import { Injectable } from "@angular/core";
+import { Injectable } from '@angular/core';
+import {AuthService, JWTApiService} from '@savvato-software/savvato-javascript-services';
 import { ReviewAttributesApiService } from "./review-attributes.api.service";
-import { ModelTransformingService } from "@savvato-software/savvato-javascript-services";
+import { ReviewAttributesPageModule } from "../review-attributes.page.module";
+import { Constants } from "../../../_constants/constants";
 
 @Injectable({
-  providedIn: "root",
+    providedIn: 'root'
 })
 export class ReviewAttributesModelService {
-  model: any = {};
-
-  constructor(
-    private _modelTransformingService: ModelTransformingService,
-    private _reviewAttributesApiService: ReviewAttributesApiService
-  ) {}
-
-  init() {
-    this._modelTransformingService.clearAllTransformers();
-
-    this._modelTransformingService.addTransformer((model, done) => {
-      this._reviewAttributesApiService
-        .getNextReviewEligiblePhrase()
-        .then((response) => {
-          model["nextReviewPhrase"] = response;
-          done();
-        });
-    });
-    this._modelTransformingService.transform(this.model);
-  }
-
-  getNextReviewEligiblePhrase() {
-    return this.model["nextReviewPhrase"];
-  }
+
+    model: any = {};
+    saveReview: any = {};
+
+    constructor(private _apiService: JWTApiService,
+                private _authService: AuthService,
+                private ReviewAttributesApiService: ReviewAttributesApiService,
+                private _constants: Constants) {
+
+    }
+
+
+    getNewPhrase() {
+        /**return new Promise((resolve, reject) => {
+           //this.model = this.ReviewAttributesApiService.getPhrase();
+          // resolve(this.model);
+         this.ReviewAttributesApiService.getPhrase().then((response) => {
+                this.model = response;
+                resolve(this.model);
+            })
+            
+        })**/
+        this.ReviewAttributesApiService.getPhrase().then((response) => {
+                //if(response != undefined){
+                    this.model = response;
+                    return this.model;
+               // }
+               // else{
+                //    throw new Error("no more phrase");
+               // }
+        })
+            
+        
+    }
+
+    saveReviewAttributes(reviewId, reasonId) {
```

Comment:
No need to take a reviewId here, since this class already has that info in its model. You could store the reasonId in the model as well, and that way you wouldn't need the reasonId parameter here either.

Reply:
I remove the reviewId parameter and use the model instead. I couldn't remove the reasonId because value changes based on if is the approve or reject method calling saveReviewAttributes method. I believe reasonId should come from the backend but there is not endpoint yet for it in the backend so that why I was using constant value for the time being.